### PR TITLE
Edit form modification

### PR DESCRIPTION
**Pull request by etiennehomer**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 1]
```diff
@@ -529,6 +541,34 @@ private Load createLoadInBusBreaker(VoltageLevel voltageLevel, LoadCreationInfos
             .add();
     }
 
+    public Mono<Void> updateGeneratorCreation(GeneratorCreationInfos generatorCreationInfos) {
+        assertGeneratorCreationInfosNotEmpty(generatorCreationInfos).subscribe();
+
+        Optional<ModificationEntity> generatorModificationEntity = this.modificationRepository.findById(generatorCreationInfos.getUuid());
+
+        if (!generatorModificationEntity.isPresent()) {
+            return Mono.error(new NetworkModificationException(CREATE_LOAD_ERROR, "Generator creation not found"));
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:550]: CREATE_GENERATOR_ERROR ?

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 2]
```diff
@@ -841,6 +903,36 @@ private Line createLine(Network network, VoltageLevel voltageLevel1, VoltageLeve
         return lineAdder.add();
     }
 
+    public Mono<Void> updateLineCreation(LineCreationInfos lineCreationInfos) {
+        assertLineCreationInfosNotEmpty(lineCreationInfos).subscribe();
+
+        Optional<ModificationEntity> lineModificationEntity = this.modificationRepository.findById(lineCreationInfos.getUuid());
+
+        if (!lineModificationEntity.isPresent()) {
+            return Mono.error(new NetworkModificationException(CREATE_LOAD_ERROR, "Line creation not found"));
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:912]: CREATE_LINE_ERROR

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 3]
```diff
@@ -1260,13 +1426,13 @@ private void changeThreeWindingsTransformerAttribute(ThreeWindingsTransformer tr
         NetworkStoreListener listener = NetworkStoreListener.create(network,
             networkUuid,
             null,
-            modificationRepository,
+                networkModificationRepository,
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:1429]: indentation

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 4]
```diff
@@ -1447,6 +1613,22 @@ private void createShuntCompensator(VoltageLevel voltageLevel, ShuntCompensatorC
         shunt.add();
     }
 
+    public Mono<Void> updateShuntCompensatorCreation(ShuntCompensatorCreationInfos shuntCompensatorCreationInfos) {
+        assertShuntCompensatorCreationInfosNotEmpty(shuntCompensatorCreationInfos).subscribe();
+
+        Optional<ModificationEntity> shuntCompensatorModificationEntity = this.modificationRepository.findById(shuntCompensatorCreationInfos.getUuid());
+
+        if (!shuntCompensatorModificationEntity.isPresent()) {
+            return Mono.error(new NetworkModificationException(CREATE_LOAD_ERROR, "Shunt compensator creation not found"));
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:1622]: CREATE_SHUNT_COMPENSATOR_ERROR ?

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 5]
```diff
@@ -97,8 +97,7 @@ public void saveModifications(UUID groupUuid, List<? extends ModificationEntity>
         var modificationGroupEntity = this.modificationGroupRepository
                 .findById(groupUuid)
                 .orElseGet(() -> modificationGroupRepository.save(new ModificationGroupEntity(groupUuid)));
-        modificationGroupEntity.getModifications().addAll(modifications);
-        this.modificationRepository.saveAll(modifications);
+        modifications.forEach(modificationGroupEntity::addModification);
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:100]: You don't need the saveAll anymore ?

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 6]
```diff
@@ -189,6 +238,13 @@ public NetworkModificationController(NetworkModificationService networkModificat
         return ResponseEntity.ok().body(networkModificationService.createVoltageLevel(networkUuid, variantId, groupUuid, voltageLevelCreationInfos));
     }
 
+    @PutMapping(value = "/voltage-levels", consumes = MediaType.APPLICATION_JSON_VALUE, produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (sBouzols) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:241]: Add the voltage-levelUuid in path for a PUT Request

**Review by sBouzols — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java (hunk 1 above)
Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java:550]: yep, fixed

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java (hunk 5 above)
Comment (AbdelHedhili) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:100]: No need indeed

**Review by sBouzols — APPROVED** [review]

**Review by AbdelHedhili — APPROVED** [review]

**Pull request merged**